src/shrutiC/NumericOperations.java

```diff
@@ -0,0 +1,81 @@
+package shrutiC;
+
+public class NumericOperations {
+
+	void displayArray(int[] numberArray) {
+		System.out.print("Array : ");
+		for (int index = 0; index < numberArray.length; index++)
+			System.out.print(numberArray[index] + " ");
+	}
+
+	void displayReverseArray(int[] numberArray) {
+		System.out.print("\nReverse Array : ");
+		for (int index = numberArray.length - 1; index >= 0; index--)
+			System.out.print(numberArray[index] + " ");
+
+	}
+
+	int findMaximumNum(int[] numberArray) {
+
+		int maxNum = numberArray[0];
+		for (int index = 0; index < numberArray.length; index++) {
+			if (numberArray[index] > maxNum) {
+				maxNum = numberArray[index];
+			}
+		}
+
+		return maxNum;
+
+	}
+
+	int findMinimumNum(int[] numberArray) {
+
+		int minNum = numberArray[0];
+		for (int index = 0; index < numberArray.length; index++) {
+			if (minNum > numberArray[index]) {
+				minNum = numberArray[index];
+			}
+		}
+
+		return minNum;
+	}
+
+	int findAverage(int[] numberArray) {
+
+		int sum = 0;
+		int avg = 0;
+		for (int index = 0; index < numberArray.length; index++)
+			sum += numberArray[index];
+
+		avg = sum / numberArray.length;
+		System.out.println("Average of all numbers in an array is : " + avg);
+		return avg;
+
+	}
+
+	void findAverageOfMinMaxNum(int[] numberArray) {
+		int avg = 0;
+		int minNum, maxNum;
+		maxNum = findMaximumNum(numberArray);
+		minNum = findMinimumNum(numberArray);
+		avg = (maxNum + minNum) / 2;
+		System.out.println("Average of min and max number in an array is : " + avg);
```

Comment:
I think we have requirement where you need to find max & min number then you need to find avg of both, user will give us array, and you need to find min & max value. and then avg of both, I hope you got my point.


You can write code like this, I remove variable and directly doing sum of min & max, no problem, if you are not comfortable to write code like below, just optimize.

int avgOfMinMax = (getMinNumber(input)+getMaxNumber(input))/2